packages/filler/src/services/CacheService.ts

```diff
@@ -0,0 +1,119 @@
+import fs from "fs"
+import path from "path"
+
+interface GasEstimateCache {
+	fillGas: string
+	postGas: string
+	timestamp: number
+}
+
+interface SwapOperationsCache {
+	calls: {
+		to: string
+		data: string
+		value: string
+	}[]
+	totalGasEstimate: string
+	timestamp: number
+}
+
+interface CacheData {
+	gasEstimates: Record<string, GasEstimateCache>
+	swapOperations: Record<string, SwapOperationsCache>
+}
+
+export class CacheService {
+	private cacheFile: string
+	private cacheData: CacheData
+	private readonly CACHE_EXPIRY_MS = 10 * 1000 // 10 seconds
+
+	constructor() {
+		this.cacheFile = path.join(process.cwd(), "cache", "filler-cache.json")
```

Comment:
If we are going to use a cache, I think it should completely be an in-memory cache, there's no advantage to storing it on disk.

Reply:
I added the cache so that executeOrder could leverage from all the contract reads, simulations already done by calculateProfitability second. Do you want me to use the in-memory option and remove the json storing logic?

Reply:
Yes, let's use in memory

Reply:
Done!